react-intl/index.d.ts

```diff
@@ -34,14 +34,16 @@ declare namespace ReactIntl {
         formats?: any;
     }
 
-    interface InjectedIntl {
+    interface InjectedIntlProps {
+      intl: {
```

Comment:
I understand you point.. but your solution isnt working. Just try it in a sample project.